sdk/storage/azure-storage-blob/samples/client_side_encryption_keyvault.py

```diff
@@ -0,0 +1,123 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: client_side_encryption_keyvault.py
+
+DESCRIPTION:
+    This sample contains code demonstrating how to conifgure the storage blob
+    service for client side encryption, storing and retrieving the key encryption key
+    (kek) from within Azure KeyVault. This sample requires a service principal be set
+    configured with access to KeyVault, and that the vault contains a 256-bit base64-
+    encoded secret named "symmetric-key". Additionally, a number of environment
+    variabled must be set. Since these often contain sensitive information, they
+    SHOULD NOT be replaced with hardcoded values in any code derived from this sample.
+
+USAGE: python client_side_encryption_keyvault.py
+"""
+
+import base64
+import os
+import sys
+import uuid
+
+from azure.common import AzureException
+
+from azure.identity import ClientSecretCredential
+
+from azure.keyvault.keys.crypto import CryptographyClient, KeyWrapAlgorithm
+from azure.keyvault.keys import KeyVaultKey
+from azure.keyvault.secrets import SecretClient
+
+from azure.storage.blob import BlobServiceClient, BlobType
+
+# Environment variable keys whic must be set to run this sample
+STORAGE_URL = 'AZURE_STORAGE_URL'
+KEYVAULT_URL = 'AZURE_KEYVAULT_URL'
+CLIENT_ID = 'AZURE_CLIENT_ID'
+CLIENT_SECRET = 'AZURE_CLIENT_SECRET'
+TENANT_ID = 'AZURE_TENANT_ID'
+
+def get_env_var(key):
+    try:
+        return os.environ[key]
+    except KeyError:
+        print('{} must be set.'.format(key))
+        sys.exit(1)
+
+def make_resource_name(prefix):
+    return '{}{}'.format(prefix, str(uuid.uuid4()).replace('-', ''))
+
+class KeyWrapper:
+    """ Class that fulfills the interface used by the storage SDK's
+        automatic client-side encyrption and decryption routines. """
+
+    def __init__(self, kek, credential):
+        self.kek = kek
+        self.kid = kek._properties._id
+        self.algorithm = KeyWrapAlgorithm.aes_256
+
+    def wrap_key(self, key):
+        crypto_client = CryptographyClient(self.kek, credential)
+        wrapped = crypto_client.wrap_key(key=key, algorithm=self.algorithm)
+        return wrapped.encrypted_key
+
+    def unwrap_key(self, key, algorithm):
```

Comment:
Constructor and `get_key_wrap_algorithm` imply a `KeyWrapper` is bound to a single algorithm, but this takes an algorithm argument?

Reply:
Yes, this is the interface defined by storage. If omitted you get an error: "unwrap_key takes three positional arguments, but two were given."

Reply:
Seems bizarre but if we must implement this interface, I'd ignore the argument and pass `self.algorithm` to `unwrap_key`. If the `KeyWrapper` performs all un/wrapping, and doesn't accept an arbitrary wrap algorithm, accepting an arbitrary unwrap algorithm is just a bug vector.

Reply:
Is this the spot? If so the comment there claims an exception is expected if the algorithm is not supported
https://github.com/Azure/azure-sdk-for-python/blob/db402b152526a41408c2e64e965a398bd893312e/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/encryption.py#L249-L251